app/grandchallenge/discussion_forums/models.py

```diff
@@ -78,6 +80,69 @@ class Meta:
     def __str__(self):
         return self.subject
 
+    def save(self, *args, **kwargs):
```

Comment:
You may want the `slug` and `parent_object` to be unique together, that way the slugs do not need to be globally unique, but unique per challenge. If you do that, you will need an extra filter on the parent object anywhere an object is looked-up with a slug in the url, so far this would be the detail and delete views.

Reply:
Good point. Not forcing global uniqueness makes sense here. I'll make the necessary changes.

Reply:
Thinking about this some more: extending this to reader studies or other parent objects can be done with the current design. The uniqueness constraint would then just need a condition added to it, and I think we could also add a check constraint to ensure that there is only either a linked_challenge or linked_reader_study, for example. It's less elegant, but possible, and in my opinion better than the GFK for performance and ease of working with the parent object relation.

Reply:
That all has me thinking that it would be good to add a constraint now that each forum MUST have a linked challenge. We don't want dangling forums, right?

Reply:
@jmsmkn could you comment on the latter (making a parent_object required for forums)?

Reply:
Turns out that I also can't add a uniqueness constraint for the related field `linked_challenge`, only fields on the model can be included in unique constraints. But then again, I think we don't need to enforce uniqueness with the parent object, but rather between slug and forum. That makes a lot of the above irrelevant.

Reply:
I have not added the constraint that a forum needs to have a linked challenge for now. If we want to add it, it needs to happen in a separate migration after both the forum model and the new field on the challenge model have been created. I can add it later, if we think it's important.

Reply:
I think there is no need to add an extra filter on any of the views, since the views already include the challenge short_name filter, and a challenge only has one forum, so that filters it down sufficiently already, right?

Reply:
You're right that it should be unique `slug` and `forum`, sorry!

> since the views already include the challenge short_name filter, and a challenge only has one forum

`TopicDetail` and `TopicDelete` does not appear to me to be filtered by forum. The only URL lookup there is the slug. The challenge is not considered by the default object resolver.

Reply:
Should the post have a title/slug? It is always viewed in context of an existing topic.

Reply:
A post can be updated though, so we need a way to refer to a single post. If there is no slug, the url for the update view would then contain the pk. If that's ok, then we could indeed do without a title and slug.

Reply:
Yes, I think that is fine.

Reply:
Looking at current post slugs, they often start with one (or multiple) "re:" (which gets prepended automatically), because it indeed doesn't really make sense to add a new title to a reply post. So yeah, we're not losing meaningful information by getting rid of the slug.